src/components/Form.tsx

```diff
@@ -1,47 +1,108 @@
-import { Fragment } from 'react';
+import { Fragment, useState, useEffect } from 'react';
+import { JuzOptions } from '@/api/murojaah';
 import { Transition, Dialog } from '@headlessui/react';
+import Select from 'react-select';
 
 interface Props {
   showForm: boolean;
   setShowForm: Dispatch<SetStateAction<boolean>>;
 }
 
 export const Form = ({ showForm, setShowForm }: Props): JSX.Element => {
+  const [selectedOption, setSelectedOption] = useState(null);
+  const [disableSave, setDisableSave] = useState(true);
+  const [cancelText, setCancelText] = useState('Cancel');
+  const [showCancelConfirmation, setShowCancelConfirmation] = useState(false);
+
+  useEffect(() => {
+    if (showForm) {
+      resetStates();
```

Comment:
can we reset form when closing the form, instead adding another `useEffect`?

Reply:
this purely for aesthetic only, so the user won't see the initial form again after clicked cancel or save button

we can achieve same result by using setTimeout(), already updated on https://github.com/RR-Ardhika/murojaah/pull/6/commits/02758916e2c35bc5d96640fa2c7e178bd353509a, thank you